### ISSUE TITLE
Use OP_RETURN to mine wtxid

### ISSUE THREAD

**Issue by kpp** [chainwayxyz/citrea#1259]
Now we put a nonce into `reveal` tx to later create a `commit` tx with the given wtxid prefix. However we can use OP_RETURN for that.

**Comment by eyusufatik** [comment]
In the reveal tx, we keep the nonce inside our script, set it to a fixed number for compatability reasons. We start using another nonce, by using the nonce after OP_RETURN in a new output we add to the reveal tx. we try to find the necessary prefix by changing the bytes in this op_return output.

so our da transaction looks like this:
|~~~~~~~~ commit tx ~~~~~~~~| ~~~~~~~~~reveal tx~~~~~~~~~~~~~~|           
commit input --> commit output --> reveal input --> reveal output
-------------------------------------------------- |--> OP_RETURN nonce

reveal input still has the script but now we don't need to clone it constantly.



so our mining looks like this:

```
script = generate_data_write_script()
nonce = 17
reveal_tx = Transaction { ins: [Input {txid: [0;32], vout:0, script}, ], outs: [Output { address, value: 546}, Output { script: "OP_RETUN" + nonce, value: 0 } ]  }

while !sign_and_hash(tx).starts_with(prefix):
    tx.outs[1].script = "OP_RETURN" + (++nonce) 
    
reveal_address = calc_reveal_address(tx)
 commit_tx = Transaction {ins: choose_utxos(), outs: [Output {address: reveal_address, amount, }, Output {address: change_address, value: change}]  }
 
 reval_tx.ins[0].tx_id = commit_tx.tx_id
```

**Comment by kpp** [comment]
Unfortunately it's not possible to eliminate ScriptBuf::clone from our mining with OP_RETURN. We will still have to clone it to create a new reveal_tx's taproot with 2 outputs: data and OP_RETURN. There is no way to pass data by ref.

When I implemented chunks, I changed the layout of tx data and put nonce in the end of the script, so no we do:
- clone script with blob
- add nonce in the end of the script
- mine wtxid and check prefix

Cloning the whole script with blob and adding a new output with OP_RETURN will not help to speed up the mining of wtxid.
